opengever/maintenance/scripts/add_missing_file_extensions.py

```diff
@@ -0,0 +1,123 @@
+"""
+This script searches documents with missing filename extension
+and adds the extension corresponding to the document mimetype.
+    bin/instance run ./scripts/add_missing_file_extensions.py
+Options:
+  -n : dry run
+  -p : path in which to search files
+"""
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from opengever.maintenance.utils import LogFilePathFinder
+from opengever.maintenance.utils import TextTable
+from plone import api
+import os
+import sys
+import transaction
+
+
+class FilenameExtensionFixer(object):
+
+    def __init__(self, options):
+        self.search_path = options.path
+        self.corrected = TextTable()
+        self.corrected.add_row(("Path", "Filename", "Mimetype", "New filename"))
+        self.skipped = TextTable()
+        self.skipped.add_row(("Path", "Filename", "Mimetype"))
+        self.registry = api.portal.get_tool("mimetypes_registry")
+
+    def fix_extensions(self):
+        for document in self.get_documents_with_missing_extension():
+            self.fix_file_extension(document)
+
+    def fix_file_extension(self, document):
+        """ Add missing extension to document filename
+        """
+        path = document.absolute_url_path()
+        filename = document.get_filename()
+
+        mimetype = None
+        try:
+            mimetype = document.get_mimetype()[0]
+            extension = self.registry.lookup(mimetype)[0].extensions[0]
+            new_filename = u"{}.{}".format(filename, extension)
+            document.get_file().filename = new_filename
+            self.corrected.add_row((path, filename, mimetype, new_filename))
+        except:
+            self.skipped.add_row((path, filename, mimetype))
+
```

Comment:
Not sure whether it is acceptable here to use a bare `except`. I could obviously handle all that a bit more cleanly, but for a maintenance script I thought that would do, especially since we list the filenames that could not be corrected.

Reply:
As discussed, I would probably just `print` the exception (plus some context, like `path`)- but other than that it's totally fine for a maintenance script. The key with no using bare `except:`'s is that exceptions might go unhandled. But appending it to `self.skipped` (and printing) *is* actually handling it in some way.